examples/compiled/vconcat_flatten.vg.json

```diff
@@ -45,7 +44,8 @@
           "type": "formula",
           "expr": "toNumber(datum[\"lc\"] && datum[\"lc\"][\"mag\"])",
           "as": "lc.mag"
-        }
+        },
+        {"type": "flatten", "fields": ["lc"], "as": ["lc"]}
```

Comment:
Yeah... some of the facet examples are broken.

Reply:
This is because of the #3832 .
`hasIntersection` returns `{}` for `{ lc: true }` & `{ ra: true, dec: true, 'lc.time': true, 'lc.mag': true }`.

Reply:
Shouldn't `lc` be in the list then?

Reply:
I had a go at that in #3890, but I think we decided to replace with `hasIntersection` with something that generated `lc` and *then* calculated the intersection instead of changing `dependentFields`.

Reply:
Ahh, I remember. @kanitw Do you know why we decided that? 

I think we either should use a different function to compute dependencies then or go the route that you initially went and add all fields in nested fields.

Reply:
@kanitw Any preferences?

Reply:
I'd say since you ran into this multiple times, let's add all fields that we depend on and deconstruct nested fields. I did something similar in https://github.com/vega/vega-lite/pull/4099